common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/WebViewAuthorizationFragment.java

```diff
@@ -276,29 +276,26 @@ public void onPermissionRequest(final PermissionRequest request) {
                 // We can only grant or deny permissions for video capture/camera.
                 // To avoid unintentionally granting requests for not defined permissions
                 // we check if the request is for camera.
-                if (!isPermissionRequestForCamera(request)) {
+                if (!CameraPermissionRequest.isValidRequest(request)) {
                     Logger.warn(methodTag, "Permission request is not for camera.");
                     request.deny();
                     return;
                 }
                 // There is a issue in ESTS UX where it sends multiple camera permission requests.
                 // So, if there is already a camera permission request in progress we handle it here.
                 if (mCameraPermissionRequest != null) {
-                    handleRepeatedRequests(request);
+                    Logger.info(methodTag, "Repeated request, granted? " + mCameraPermissionRequest.isGranted());
+                    handleRepeatedCameraRequests(request);
                     return;
                 }
                 Logger.info(methodTag, "New camera request.");
-                mCameraPermissionRequest = request;
-                if (isCameraPermissionGranted()) {
+                mCameraPermissionRequest = new CameraPermissionRequest(request);
+                if (isAppCameraPermissionGranted()) {
                     Logger.info(methodTag, "Camera permission already granted.");
-                    acceptCameraRequest();
-                } else if (shouldShowRequestPermissionRationale(Manifest.permission.CAMERA)) {
-                    Logger.info(methodTag, "Show camera rationale.");
                     showCameraRationale();
```

Comment:
are we supposed to ask just for the permission (1 dialog) or both rationale + permission (2 dialogs) every time?

Reply:
also... if isCameraPermissionGranted == false, then we don't show the rationale? we just pop the camera request activity?

Reply:
would you mind adding that as a comment here?

Reply:
would you mind adding that as a comment here?